### PR TITLE
Rename liveLogger to be terminallogger

### LINKED ISSUE #8915: [Bug]: References to some mysterious "LiveLogger" still exist in MSBuild
### Issue Description

We still refer to LiveLogger in places in our code like [here](https://github.com/dotnet/msbuild/blob/2c5c951d631c0dc58a1c7a2d0780c475a91a81e6/src/MSBuild/XMake.cs#L3367). We should probably replace those with TerminalLogger.

### Steps to Reproduce

Open MSBuild
Look at code

### Expected Behavior

No LiveLogger, FancyLogger, CurrentLogger, FutureLogger, ModernLogger, BetterLogger references

### Actual Behavior

LiveLogger references found!

### Analysis

_No response_

### Versions & Configurations

_No response_

### PR DESCRIPTION
**Pull request by GangWang01**
Fixes #8915

### Context
For live logger it has a couple of previous names mentioned in the original issue. Now the agreed name is terminal logger.

### Changes Made
Rename live logger to terminal logger in the source code including comments and doc.

### CODE REVIEW

File: src/MSBuild/CommandLineSwitches.cs [hunk 1]
```diff
@@ -246,7 +246,7 @@ private struct ParameterizedSwitchInfo
             new ParameterizedSwitchInfo(  new string[] { "fileloggerparameters7", "flp7" },     ParameterizedSwitch.FileLoggerParameters7,      null,                           false,          "MissingFileLoggerParameterError",     true,   false),
             new ParameterizedSwitchInfo(  new string[] { "fileloggerparameters8", "flp8" },     ParameterizedSwitch.FileLoggerParameters8,      null,                           false,          "MissingFileLoggerParameterError",     true,   false),
             new ParameterizedSwitchInfo(  new string[] { "fileloggerparameters9", "flp9" },     ParameterizedSwitch.FileLoggerParameters9,      null,                           false,          "MissingFileLoggerParameterError",     true,   false),
-            new ParameterizedSwitchInfo(  new string[] { "livelogger", "ll", "terminallogger", "tl" }, ParameterizedSwitch.LiveLogger,          null,                           true,          null,                                  true,   true),
```
Comment (rainersigwald) [inline src/MSBuild/CommandLineSwitches.cs:249]: It's a breaking change to stop supporting the `-livelogger` and `-ll` command line switches. I don't think I'm willing to do that.

File: src/MSBuild/XMake.cs [hunk 2]
```diff
@@ -2514,69 +2514,69 @@ private static bool GatherAutoResponseFileSwitchesFromFullPath(string autoRespon
...
-                if (string.IsNullOrEmpty(liveLoggerArg))
+                if (string.IsNullOrEmpty(terminalloggerArg))
                 {
-                    liveLoggerArg = "auto";
+                    terminalloggerArg = "auto";
                 }
             }
             else
             {
-                liveLoggerArg = Environment.GetEnvironmentVariable("MSBUILDLIVELOGGER");
```
Comment (rainersigwald) [inline src/MSBuild/XMake.cs:2537]: Please continue to support the `MSBUILDLIVELOGGER` approach. It's fine to add and change all the docs to `MSBUILDTERMINALLOGGER`, though.

Comment (GangWang01) [reply src/MSBuild/XMake.cs:2537]: If both `MSBUILDLIVELOGGER` and `MSBUILDTERMINALLOGGER` are configured, which one takes precedence?

Comment (rainersigwald) [reply src/MSBuild/XMake.cs:2537]: The new/more correct name, `MSBUILDTERMINALLOGGER`.

File: src/MSBuild/CommandLineSwitches.cs (hunk 1 above)
Comment (GangWang01) [reply src/MSBuild/CommandLineSwitches.cs:249]: The switches for live logger were added back.

File: src/MSBuild/XMake.cs (hunk 2 above)
Comment (GangWang01) [reply src/MSBuild/XMake.cs:2537]: added the support `MSBUILDLIVELOGGER ` back.

**Review by YuliiaKovalova — APPROVED** [review]
Looks good to me!

**Pull request merged**